apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/TimestampBoundTests.cs

```diff
@@ -0,0 +1,157 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Cloud.ClientTesting;
+using Google.Cloud.Spanner.V1;
+using Google.Protobuf.WellKnownTypes;
+using System;
+using Xunit;
+using static Google.Cloud.Spanner.V1.TransactionOptions.Types;
+
+namespace Google.Cloud.Spanner.Data.Tests
+{
+    public class TimestampBoundTests
+    {
+        public static TheoryData<TimestampBound> SampleBounds = new TheoryData<TimestampBound>
+        {
+            TimestampBound.OfMaxStaleness(TimeSpan.FromHours(1)),
+            TimestampBound.OfExactStaleness(TimeSpan.FromHours(1)),
+            TimestampBound.OfMinReadTimestamp(new DateTime(1999, 12, 31, 3, 33, 33, DateTimeKind.Utc)),
+            TimestampBound.OfReadTimestamp(new DateTime(1999, 12, 31, 3, 33, 33, DateTimeKind.Utc)),
+            TimestampBound.Strong
+        };
+
+        [Theory]
+        [MemberData(nameof(SampleBounds))]
+        public void Base64RoundTrip(TimestampBound bound)
+        {
+            Assert.Equal(bound, TimestampBound.FromBase64String(bound.ToBase64String()));
+        }
+
+        [Fact]
+        public void ToTransactionOptions_Strong()
+        {
+            AssertProtoConversion(new ReadOnly { Strong = true }, TimestampBound.Strong);
+        }
+
+        [Fact]
+        public void ToTransactionOptions_OfMaxStaleness()
+        {
+            var bound = TimestampBound.OfMaxStaleness(TimeSpan.FromSeconds(5));
+            AssertProtoConversion(new ReadOnly { MaxStaleness = new Duration { Seconds = 5 } }, bound);
+        }
+
+        [Fact]
+        public void ToTransactionOptions_OfExactStaleness()
+        {
+            var bound = TimestampBound.OfExactStaleness(TimeSpan.FromSeconds(5));
+            AssertProtoConversion(new ReadOnly { ExactStaleness = new Duration { Seconds = 5 } }, bound);
+        }
+
+        [Fact]
+        public void ToTransactionOptions_OfMinReadTimestamp()
+        {
+            var bound = TimestampBound.OfMinReadTimestamp(new DateTime(1970, 1, 1, 0, 0, 10, DateTimeKind.Utc));
+            AssertProtoConversion(new ReadOnly { MinReadTimestamp = new Timestamp { Seconds = 10 } }, bound);
+        }
+
+        [Fact]
+        public void ToTransactionOptions_OfExactTimestamp()
+        {
+            var bound = TimestampBound.OfReadTimestamp(new DateTime(1970, 1, 1, 0, 0, 10, DateTimeKind.Utc));
+            AssertProtoConversion(new ReadOnly { ReadTimestamp = new Timestamp { Seconds = 10 } }, bound);
+        }
+
+        [Theory]
+        [InlineData(DateTimeKind.Local)]
+        [InlineData(DateTimeKind.Unspecified)]
+        public void Timestamp_NonUtcIsInvalid(DateTimeKind kind)
+        {
+            var timestamp = new DateTime(2000, 1, 1, 0, 0, 0, kind);
+            Assert.Throws<ArgumentException>(() => TimestampBound.OfMinReadTimestamp(timestamp));
```

Comment:
nit: Is this the same line twice?

Reply:
Typo - second one should be OfReadTimestamp